docs/pages/eas-workflows/jobs.mdx

```diff
@@ -3,105 +3,137 @@ title: Jobs
-      flow_path: string | string[] # string is a file path from the project root
+e2e-test:
+  name: Run E2E Tests
+  type: maestro
+  params:
+    build_id: string # required - usually ${{ needs.build.outputs.build_id }}
+    flow_path: string | string[] # required - path(s) to Maestro test file(s)
 ```
 
-## Custom jobs
-
-Custom jobs are jobs that you write yourself. They are useful for any use case that isn't met by the pre-packaged jobs.
-
-```yaml .eas/workflows/custom-job.yaml
-jobs:
-  custom-job:
-    type: custom # default: undefined
-    steps:
-      - name: Run a script
-        run: echo "Hello, world!"
+## Get a specific build
+
+Gets an existing build based supplied params:
+
+```yaml .eas/workflows/get-build.yaml
+get-build:
+  name: Get Build
+  type: get-build
+  params:
+    platform: ios | android # required
+    profile: string # optional
+    app_identifier: string # optional
+    app_version: string # optional
+    app_build_version: string # optional
+    channel: string # optional
+    distribution: store | internal | simulator # optional
+    fingerprint_hash: string # optional
+    git_commit_hash: string # optional
+    runtime_version: string # optional
+    sdk_version: string # optional
+    simulator: boolean # optional
 ```
 
-Within custom jobs, you can use our [built-in functions](/custom-builds/schema/#built-in-eas-functions), like `eas/checkout`, `eas/use_npm_token` to interact with the project. These need to be prefixed with the `uses:` keyword.
+## Deploy your app
 
-```yaml .eas/workflows/custom-job.yaml
-jobs:
-  custom-job:
-    type: custom
-    steps:
-      - uses: eas/checkout
+Deploy to EAS Hosting:
 
-      - uses: eas/install_node_modules
+```yaml .eas/workflows/deploy.yaml
+deploy:
+  name: Deploy App
+  type: deploy
+  environment: production | preview | development # optional
+  params:
+    alias: string # optional - custom alias for the new deployment
+    prod: boolean # optional - create a production deployment
+```
 
-      - name: Say hello!
-        run: echo "Hello world!"
+## Custom jobs
 
-      - name: Multiline script
-        run: |
-          CURRENT_DATE=$(date)
-          echo "It is $CURRENT_DATE"
+Run custom code during a workflow:
+
+```yaml .eas/workflows/custom-job.yaml
+custom-job:
+  name: Custom Task
+  runs_on: linux-medium | linux-large | macos-medium | macos-large | linux-medium-nested-virtualization | linux-large-nested-virtualization # optional, default: linux-medium
+  defaults:
+    run:
+      working_directory: string # optional - relative to app base directory or absolute from repo root
+  outputs:
+    key: value # optional - mapping from steps outputs to job outputs
+  steps:
+    # Shell command step
+    - name: Run Command # optional
+      id: step-id # optional - useful for referencing outputs
+      run: echo "Hello World"
+      working_directory: string # optional
+      outputs: # optional
+        - name: output-name
+          required: boolean
```

Comment:
Hmm, I know we do support them, but I wish we wouldn't. This is a remainder of Custom Builds which were much cleverer about outputs, they supported different types, validated, stopped workflow if a required output wasn't defined, etc. I'm not sure we should do this for workflows, so I'd rather not advertize this.

The only problem with setting an output not defined in `step.outputs` is that worker will print a warning ([as experienced by James](https://exponent-internal.slack.com/archives/C06EFBQK3B7/p1737419576613249)). Planning to get rid of the warning.